tornado/test/web_test.py

```diff
@@ -277,6 +277,7 @@ def test_cookie_special_char(self):
                 # ('foo=a\\073b', 'a;b'),  # even encoded, ";" is a delimiter
                 ('foo="a\\073b"', 'a;b'),
                 ('foo="a\\"b"', 'a"b'),
+                ('foo@bar=1; foo=1', '1')
```

Comment:
The change description includes a more complex test case that should be included here as well.